app/controllers/auth/FMNAuth.scala

```diff
@@ -165,7 +165,7 @@ trait FMNAuth extends AuthorisedFunctions with Logging {
             .asInstanceOf[Request[A]]
 
           val authenticatedRequest = AuthContext[A](
-            trustedHelper.fold(NationalInsuranceNumber(nino))(helper => NationalInsuranceNumber(helper.principalNino)),
+            trustedHelper.fold(NationalInsuranceNumber(nino))(helper => NationalInsuranceNumber(helper.principalNino.getOrElse(nino))),
```

Comment:
This would be simpler if have separate case for Some(trustedHelper) then would not need fold etc